spec/requests/api/v1/dashboard/dashboards_spec.rb

```diff
@@ -0,0 +1,88 @@
+    position_title: "Jr. CTO",
+    date_applied: "2024-10-31",
+    status: 1,
+    notes: "Fingers crossed!",
+    job_description: "Looking for Turing grad/jr dev to be CTO",
+    application_url: "www.example1.com",
+    contact_information: "boss@example.com",
+    company_id: test_company.id,
+    user_id: user.id) }
+
+  let(:job_application2) {JobApplication.create!(
+    position_title: "Frontend Developer",
+    date_applied: "2024-11-01",
+    status: 0,
+    notes: "Excited about this opportunity!",
+    job_description: "Frontend Developer role with React expertise",
+    application_url: "www.frontend.com",
+    contact_information: "hiring@example.com",
+    company_id: test_company.id,
+    user_id: user.id) }
+
+  let(:contact_params) { Contact.create!(
+    first_name: "Josnny",
+    last_name: "Smsith",
+    company_id: test_company.id,
+    email: "jonny@gmail.com",
+    phone_number: "555-785-5555",
+    notes: "Good contact for XYZ",
+    user_id: user.id) }
+
+    before(:each) do
+      post "/api/v1/sessions", params: { email: user.email, password: user.password }, as: :json
+      expect(response).to have_http_status(:ok)
+      @token = JSON.parse(response.body)["token"]
+
+      post "/api/v1/users/#{user.id}/companies", params: :test_company,
+        headers: { "Authorization" => "Bearer #{@token}" }, as: :json
+      post "/api/v1/users/#{user.id}/job_applications", params: job_application1,
+        headers: { "Authorization" => "Bearer #{@token}" }, as: :json
+      post "/api/v1/users/#{user.id}/job_applications", params: job_application2,
+        headers: { "Authorization" => "Bearer #{@token}" }, as: :json
+      post "/api/v1/users/#{user.id}/contacts", params: contact_params,
+        headers: { "Authorization" => "Bearer #{@token}" }, as: :json
+      end
+
+    it "will display a user's dashboard and have correct attributes" do
+      get "/api/v1/users/#{user.id}/dashboard",
+        headers: { "Authorization" => "Bearer #{@token}" },as: :json
+
+      expect(response).to have_http_status(:ok)
+
+      json = JSON.parse(response.body, symbolize_names: true)[:data]
+
+      expect(json[:id]).to eq(user.id.to_s)
+      expect(json[:type]).to eq("dashboard")
+      expect(json[:attributes][:name]).to eq("Testy Tim")
+      expect(json[:attributes][:email]).to eq("testytim420@email.com")
+
+      dashboard_summary = json[:attributes][:dashboard][:weekly_summary]
+      
+      expect(dashboard_summary[:job_applications].count).to eq(2)
+      expect(dashboard_summary[:job_applications].first[:position_title]).to eq("Jr. CTO")
+      expect(dashboard_summary[:job_applications].last[:position_title]).to eq("Frontend Developer")
+      expect(dashboard_summary[:companies].first[:name]).to eq("testCO")
+      expect(dashboard_summary[:companies].first[:website]).to eq("testcompany.com")
+      expect(dashboard_summary[:contacts].count).to eq(1)
+    end
+  end
+end
```

Comment:
Idk if you have this already but I know you asked for a edge case. Maybe a sad path if the user doesn't have resources?  Couldn't think of anything else

Reply:
```
{
    "data": {
        "id": "6",
        "type": "dashboard",
        "attributes": {
            "id": 6,
            "name": "Dolly Parton",
            "email": "dollyP@email.com",
            "dashboard": {
                "weekly_summary": {
                    "job_applications": [],
                    "contacts": [],
                    "companies": []
                }
            }
        }
    }
}
```

Mrs. Parton has no resources so nothings showing up.  

Although I did discover something interesting that might be an issue...
So in Postman, I was testing this with the seeded "Danny Devito"/user5.  I had previously created contacts, job apps and companies with this user to see my json response. Here:

```
{
    "data": {
        "id": "5",
        "type": "dashboard",
        "attributes": {
            "id": 5,
            "name": "Danny DeVito",
            "email": "danny_de@email.com",
            "dashboard": {
                "weekly_summary": {
                    "job_applications": [
                        {
                            "id": 1,
                            "position_title": "Jr. CTO",
                            "date_applied": "2024-10-31",
                            "status": 1,
                            "notes": "Fingers crossed!",
                            "job_description": "Looking for Turing grad/jr dev to be CTO",
                            "application_url": "www.example.com",
                            "contact_information": "boss@example.com",
                            "created_at": "2024-12-14T17:20:41.979Z",
                            "updated_at": "2024-12-14T17:20:41.979Z",
                            "company_id": 1,
                            "user_id": 5
                        },
                        {
                            "id": 2,
                            "position_title": " CTO",
                            "date_applied": "2024-10-31",
                            "status": 2,
                            "notes": "Fingers crossed!",
                            "job_description": "Looking for Turing grad/jr dev to be CTO",
                            "application_url": "www.testexample.com",
                            "contact_information": "boss1@example.com",
                            "created_at": "2024-12-14T17:37:28.465Z",
                            "updated_at": "2024-12-14T17:37:28.465Z",
                            "company_id": 2,
                            "user_id": 5
                        }
                    ],
                    "contacts": [
                        {
                            "id": 1,
                            "first_name": "Jonny",
                            "last_name": "Smith",
                            "email": "jonny@gmail.com",
                            "phone_number": "555-785-5555",
                            "notes": "Good contact for XYZ",
                            "created_at": "2024-12-14T17:55:21.875Z",
                            "updated_at": "2024-12-14T17:55:21.875Z",
                            "user_id": 5,
                            "company_id": 1
                        },
                        {
                            "id": 2,
                            "first_name": "Josnny",
                            "last_name": "Smsith",
                            "email": "jonny@gmail.com",
                            "phone_number": "555-785-5555",
                            "notes": "Good contact for XYZ",
                            "created_at": "2024-12-15T01:57:14.557Z",
                            "updated_at": "2024-12-15T01:57:14.557Z",
                            "user_id": 5,
                            "company_id": 1
                        }
                    ],
                    "companies": [
                        {
                            "id": 1,
                            "user_id": 5,
                            "name": "New Company",
                            "website": "www.company.com",
                            "street_address": "123 Main St",
                            "city": "New York",
                            "state": "NY",
                            "zip_code": "10001",
                            "notes": "This is a new company.",
                            "created_at": "2024-12-14T17:20:10.909Z",
                            "updated_at": "2024-12-14T17:20:10.909Z"
                        },
                        {
                            "id": 2,
                            "user_id": 5,
                            "name": "New Company1",
                            "website": "www.company1.com",
                            "street_address": "1231 Main St",
                            "city": "New York",
                            "state": "NY",
                            "zip_code": "10001",
                            "notes": "This is a new company1.",
                            "created_at": "2024-12-14T17:37:24.153Z",
                            "updated_at": "2024-12-14T17:37:24.153Z"
                        }
                    ]
                }
            }
        }
    }
}
```

To test out your comment here, I created a new session with the seeded "Dolly Parton" user and generated a token, but did not create anything else(job apps, etc.) with this user.  That is the JSON at the very top you're seeing.

The problem is that the user id ```GET http://localhost:3001/api/v1/users/6/dashboard```(which is the endpoint above) is completely arbitrary.  As long as your using a users bearer token, a user can be ```/users/10000/``` or ```users/12312312/``` for all that matters, and you're still gonna get a result completely based on what the JWT bearer is.

```GET http://localhost:3001/api/v1/users/10000/dashboard```  will display the same response as the top... 

Should we refactor routes? Is it a big deal?  JWT seems to be taking the reigns.

Reply:
We might have to refactor the routes but we refactored once already when we voted in class. We might have to get together and look at it again. But if someone can figure something out. It would be great.

Reply:
Yeah, it seems that :user_id in the routes is redundant/YAGNI as long as were using token-based auth to our BE API.  So I suspect that we could possibly refactor our **routes.rb**(just changing `resources` to `resource`)for simplicity.

Reply:
Pulled down to my local and played around with it. Looks like we might have to add a check to the controllers to make sure the params user_id matches the current user id. I might be wrong, if we could get more eyes on the issue and hopefully get it working again.

Reply:
If we keep the routes the way they are, and have user_id in them, we could add a validation step (as Melchor has spoken on) to make sure the user_id matches the ID of the authenticated user (from the token). This would ensure the user_id matches the token’s user_id, but it does add more complexity compared to simply removing user_id from the routes. We could add this validation to the ApplicationController so that all of our controllers inherit this method, put it in what we already have there now.

I see what you and Melchor are seeing: that user_id is not necessary here because we are getting the correct user through the token. The authenticated user is determined by the token, and the token knows the user_id. I ran into this issue when working with the frontend on creating a contact from the contacts index page. It didn’t matter what user_id I was using in the route—the token was determining what user I was actually getting contact information from, which wasn't surprising.

If we refactored to take user_id out of our routes, we could rely solely on the JWT token for user identification. Resources could be changed to resource, and there is no YAGNI. This approach simplifies the routing and shows that the current user is always determined by the token, not a route parameter. However, we had a discussion about this already, and the group decided against this approach. Unless we want to have another discussion on this, I think we have to do the check in the application controller--this would at least keep things DRY-er.

Reply:
> If we keep the routes the way they are, and have user_id in them, we could add a validation step (as Melchor has spoken on) to make sure the user_id matches the ID of the authenticated user (from the token). This would ensure the user_id matches the token’s user_id, but it does add more complexity compared to simply removing user_id from the routes.

Totally agree on added complexity.

> If we refactored to take user_id out of our routes, we could rely solely on the JWT token for user identification. Resources could be changed to resource, and there is no YAGNI. This approach simplifies the routing and shows that the current user is always determined by the token, not a route parameter. However, we had a discussion about this already, and the group decided against this approach. 

Open to discussion in the future.  Perhaps more people will experience this first-hand, as I've had.  Yet, as PM stated, this could add tech debt.  Can't stop thinking about it though 🤯

Reply:
Listening to PM, we can add this to a new ticket, to either revisit this discussion OR we keep as is and add validation method to the application controller to have user_id match with token user_id. Merging this PR now, without that, feels not as complete, however, it seems all of this great work you have done, adding the policies, getting the overwhelming majority of tests to pass, and that fact that this is working, albeit the user_id in the path being wonky, this seems like a good PR to merge now and add this other issue in a separate ticket.